src/main/sensors/gyro.c

```diff
@@ -589,9 +615,10 @@ void gyroUpdateSensor(gyroSensor_t *gyroSensor)
 
     if (isGyroSensorCalibrationComplete(gyroSensor)) {
         // move gyro data into 32-bit variables to avoid overflows in calculations
-        gyroSensor->gyroDev.gyroADC[X] = (int32_t)gyroSensor->gyroDev.gyroADCRaw[X] - (int32_t)gyroSensor->gyroDev.gyroZero[X];
-        gyroSensor->gyroDev.gyroADC[Y] = (int32_t)gyroSensor->gyroDev.gyroADCRaw[Y] - (int32_t)gyroSensor->gyroDev.gyroZero[Y];
-        gyroSensor->gyroDev.gyroADC[Z] = (int32_t)gyroSensor->gyroDev.gyroADCRaw[Z] - (int32_t)gyroSensor->gyroDev.gyroZero[Z];
+    	// move 16-bit gyro data into 32-bit variables to avoid overflows in calculations
+    	for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) // assume unroll
```

Comment:
Change to just apply this on `Z` axis. It is not required on `X` and `Y` axes and in gyro update every instruction cycle counts. (At 32kHz gyro we have 31 microseconds for the entire pid loop).

Reply:
OK. I will assume the gcc will still  inline SlewLimiter in mods and flatten loop. I will leave the 1<<14 as a constant sub expression but maybe it should be set from config?